solr/core/src/java/org/apache/solr/handler/IndexFetcher.java

```diff
@@ -677,12 +677,12 @@ IndexFetchResult fetchLatestIndex(boolean forceReplication, boolean forceCoreRel
         markReplicationStop();
         return successfulInstall ? IndexFetchResult.INDEX_FETCH_SUCCESS : IndexFetchResult.INDEX_FETCH_FAILURE;
       } catch (ReplicationHandlerException e) {
-        log.error("User aborted Replication");
+        log.error("User aborted Replication", e);
         return new IndexFetchResult(IndexFetchResult.FAILED_BY_EXCEPTION_MESSAGE, false, e);
       } catch (SolrException e) {
         throw e;
       } catch (InterruptedException e) {
-        throw new InterruptedException("Index fetch interrupted");
+        throw new InterruptedException("Index fetch interrupted", e);
```

Comment:
Locally this does not compile for me on Java 11, maybe instead this?
```suggestion
        throw (InterruptedException)(new InterruptedException("Index fetch interrupted").initCause(e));
```

Reply:
You're absolutely right! Way better fix!